src/contract/event.h

```diff
@@ -96,30 +118,17 @@ class Event {
       }
     }
 
+    // Copy constructor
+    Event(const Event&) = default;
```

Comment:
You don't really need to do this. Those are required in the presence of user-defined destructor, copy constructor, move constructor, or the equivalent assign operators. Looks like you didn't make any of those. Right? Or am I missing something?

Check **rule of five** here: https://en.cppreference.com/w/cpp/language/rule_of_three